src/elliptic.jl

```diff
@@ -0,0 +1,234 @@
+# References:
+#  [1] Abramowitz, Stegun: Handbook of Mathematical Functions (1965)
+#  [2] ellipjc.m in Toby Driscoll's Schwarz-Christoffel Toolbox
+
+
+#------------------------------------------------
+# Descending and Ascending Landen Transformation
+
+descstep(m) = m/(1+sqrt(1-m))^2
+
+@generated function shrinkm(m,::Val{N}) where {N}
+    # [1, Sec 16.12]
+    quote
+        f = one(m)
+        Base.Cartesian.@nexprs $N i->begin
+            k_i = descstep(m)
+            m = k_i^2
+            f *= 1+k_i
+        end
+        return (Base.Cartesian.@ntuple $N k), f, m
+    end
+end
+
+function ellipj_smallm(u,m)
+    # [1, Sec 16.13]
+    sn = sin(u) - m*(u-sin(u)*cos(u))*cos(u)/4
```

Comment:
at the moment, Julia doesn't yet know that `sin`/`cos` are pure, so you may want to define these as constants. Also, on 0.7 we can use `sincos`, so maybe something like
```
if VERSION < v"0.7-"
    sinu = sin(u)
    cosu = cos(u)
else
    sinu, cosu = sincos(u)
end
```

Reply:
All right, I'll fix this. The basic arithmetic (`+-*/`) is pure though, right? How about `sqrt`?

Reply:
Yes, they're all fine.